### PR TITLE
Move the location of the words executable and support files from the

### PR DESCRIPTION
**Pull request by paladine**
Move the location of the words executable and support files from the data directory to the cache directory. This means it won't get backed-up automatically each night.

Run bmgr backupnow --all to see how much data it currently uploads -
it's a lot!

Also did some cursory code cleanups.
 * Strengthened visibility of many functions
 * Properly closed Input/Output streams in case of failures
 * Avoiding copying files that have already been copied

### CODE REVIEW

File: src/main/java/com/ids1024/whitakerswords/WhitakersWords.java [hunk 1]
```diff
@@ -33,60 +34,102 @@
...
+
+        getFile(WORDS_EXECUTABLE).setExecutable(true);
+    }
+
+    private void copyFile(String filename, byte[] buffer) throws IOException {
+        InputStream ins = null;
+        FileOutputStream fos = null;
+        File outputFile = getFile(filename);
+        // if the file already exists, don't copy it again 
+        if (outputFile.exists()) {
```
Comment (ids1024) [inline src/main/java/com/ids1024/whitakerswords/WhitakersWords.java:63]: Yeah, I had been meaning to make it not extract all the files every time the app runs. However, it needs some way to allow updating these files when the app updates.

Perhaps it should save the current version of whitakers-words-android to a file in the cache directory, and update the files if it differs? Or is there a better way to do this?

**Comment by ids1024** [comment]
Thanks; I haven't done much programming in Java (or for Android), so I hadn't though to add private, or several other improvements you made.

It should probably also check the data directory and clear it, so that users can upgrade from older versions without either wasting space or requiring the user to manually clear the data.

**Comment by paladine** [comment]
Good idea. I can add some code to wipe out the data directory.

There are several ways to ensure up to date files when a new version rolls
out. I'll think it over and see which one is best/simplest.

In the past I've had directories associated with app version, like
/cache/1/... And /cache/2 etc. The version is pulled from the manifest.
That might be the best route

**Comment by ids1024** [comment]
I've just realized that this has broken the settings, which work by creating a `WORD.MOD` file in the same directory as the `words` binary.

**Comment by paladine** [comment]
Moving settings into SharedPreferences should resolve that

**Comment by ids1024** [comment]
Settings does use `SharedPreferences`, but it needs to write the settings to a file for `words` to read them.

The code in `WhitakersSettings.java` needs to be changed to write to the cache directory, and `copyFiles()`  needs to also update the settings. I think that is all.

I'll try making these changes now.

**Comment by ids1024** [comment]
Ok, 621ae37512f52af19e75ea579f313d34ea104dae should fix it.

**Pull request merged**